### DISCUSSION TITLE
Explore graph UI: allow user to configure max time delta for "connect gaps" behavior

### DISCUSSION THREAD

**Discussion by ethirolle** [grafana/grafana#38231] [Dashboards, Panels & Visualizations]
**What would you like to be added**:
Allow user to configure the max time delta for the "connect gaps" behavior for the graph in Explore UI.

**Why is this needed**:
Customer has noticed that the Explore view's behavior has changed recently: the graph panel in Explore seems to always connect lines between the data points, even when the data points are very far apart. They say:
>This makes it next to impossible to figure out when the condition (`avg(go_memstats_alloc_bytes) > 30000000`) was actually true.
>The graph panel is much saner. It looks like points are only connected to other points within some certain timedelta.

and

>Interpolation in the Explore view is a surprisingly big (negative) impact on the tool's usability. Creates the impression of data that is not there. I've been fielding confusion from internal users.

I don't think the developers will want to revert this behavior in Explore graph, because (according to the PR where this fix was made, https://github.com/grafana/grafana/pull/30707), I think the change was made in part to prepare for the new high-performance "time series" panel that was introduced in beta in 7.4 (https://grafana.com/docs/grafana/latest/panels/visualizations/time-series/#time-series-panel), and will be formally released in Grafana v8.

The solution would be to provide the user some way of configuring this behavior in Explore UI.

Some thoughts from internal convo with @leeoniya :

>the best route is inserting null values between adjacent points that have a time delta over a given threshold. if this cannot be done on the data source side, it can maybe be done in a transform.

>i guess the point is more that we should do the null insertion into the dataframe at some point before it gets to the chart. how and when that should happen is :shrug:. i don't think the customer needs to do this, but it might be useful to provide some simple knob that controls the max time delta for "connect gaps" behavior.

**Screenshots:**
Query in Explore UI:
![image](https://user-images.githubusercontent.com/10763196/109540643-c7bf1480-7a90-11eb-92aa-d27eb9ed456e.png)

Same query in a dashboard panel:
![image](https://user-images.githubusercontent.com/10763196/109540685-d3aad680-7a90-11eb-9a25-f71f533e63d6.png)

Per customer:
>The graph panel is much saner. It looks like points are only connected to other points within some certain timedelta.

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
